feat: pokemon details action

Miqqy0069 opened the pull request:
-tried before and after for GetPokemonDetails Action
-added reduce for pokemonDetails

Before 
![image_720](https://user-images.githubusercontent.com/75653480/226889999-07ca4ce5-070b-4937-9896-e8ba6cc478d1.png)

Reduce
![image_720](https://user-images.githubusercontent.com/75653480/226890036-2a689e7e-d941-4a64-925a-7e3f64193eb6.png)

After 
![image_720](https://user-images.githubusercontent.com/75653480/226890056-9d522666-48bf-434d-bc85-42d3f9d75398.png)

Analyze 
![image_360](https://user-images.githubusercontent.com/75653480/226890086-d64a6831-cf9d-44df-b0a4-e11139c7f4b4.png)

Dart Format 
![image_720](https://user-images.githubusercontent.com/75653480/226890113-fb6d7bb5-cd10-4d0a-98d0-ce3c2db64531.png)

Working App
![image_480](https://user-images.githubusercontent.com/75653480/226890197-4cc71aaf-564a-42eb-a981-ed0e1426947c.png)

Miqqy0069 committed:
feat: added pokemon details action
-tried before and after for GetPokemonDetails Action
-added reduce for pokemonDetails

Miqqy0069 committed:
feat: pokemon details action
- fixed issues on analyze
- formatted dart

shawn-ffufm commented on lib/state/app_state.dart:
```diff
@@ -10,6 +11,8 @@ class AppState with _$AppState {
   factory AppState({
     @Default('') String sample,
     @Default(<Pokemon>[]) List<Pokemon> pokemons,
+    @Default(PokemonDetails(height: 0, weight: 0, baseExperience: 0, abilities: [], moves: [], stats: []))
```
make this nullable instead with null as its default value 🙂

Miqqy0069 replied:
Resolved po sir

shawn-ffufm reviewed (changes requested):
@Miqqy0069 see minor feedback 🙂

Miqqy0069 committed:
feat: pokemon details action
- made PokemonDetails nullable with a null as a default value
- ran analyze again with 0 issues
- re ran dart format with 0 changes

shawn-ffufm commented on lib/state/action/pokemon_actions.dart:
```diff
@@ -13,3 +13,17 @@ class GetPokemonsAction extends ReduxAction<AppState> {
+
+/// Getting the pokemon Details from pokemon api
+class GetPokemonDetailsAction extends ReduxAction<AppState> {
+  final String pokemonName;
```
move this below of your constructor 🙂

Miqqy0069 replied:
Resolved po

shawn-ffufm reviewed (changes requested):
see minor feedback 🙂

Miqqy0069 committed:
feat: pokemon details action
(pr feedback)
- moved pokemonName below the constructor of GetPokemonDetailsAction

shawn-ffufm commented on lib/features/home_page.dart:
```diff
@@ -8,6 +8,7 @@ class HomePage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     StoreProvider.dispatch(context, GetPokemonsAction());
```
add a todo comment that these will be removed later 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/features/home_page.dart:
```diff
@@ -8,6 +8,7 @@ class HomePage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     StoreProvider.dispatch(context, GetPokemonsAction());
+    StoreProvider.dispatch(context, GetPokemonDetailsAction(pokemonName: 'ditto'));
```
kindly place the ditto string in a constant and have a todo comment as well 🙂

Miqqy0069 replied:
resolved

shawn-ffufm reviewed (changes requested):
see pr feedback

Miqqy0069 committed:
feat: pokemon details action
(pr feedback)
- added pokemonName(ditto) to string constant
- added Todo on dispatch for both getpokemon and getpokemondetails

shawn-ffufm commented on lib/utilities/string_constants.dart:
```diff
@@ -3,3 +3,5 @@ const homePageText = 'Home Page';
 const offset = '0';
 const limit = '151';
+
+const pokemonDitto = 'ditto';
```
you forgot the todo comment for this 😛

Miqqy0069 replied:
resolved

shawn-ffufm reviewed (changes requested):
see minor feedback

Miqqy0069 committed:
feat: pokemon details action
(pr feedback)
- added a TODO on string constants for pokemonName to be removed later
